### ISSUE TITLE
Dual Certificate in Ingress gateway

### ISSUE THREAD

**Issue by rasvi24** [istio/istio#36181]
**Describe the feature request**
Gateway needs to support dual certificate. Nginx and Envoy supports this. Currently Istio supports specifying only one TLS certificate. It is helpful in RSA/ECDSA dual certs. https://github.com/envoyproxy/envoy/pull/16605

**Describe alternatives you've considered**
I don't see an alternative.
Edit: is it possible with envoy filter specifying secret name without mounting it explicitly?

**Affected product area (please put an X in all that apply)**

[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Affected features (please put an X in all that apply)**

[ ] Multi Cluster
[ ] Virtual Machine
[ ] Multi Control Plane

**Additional context**

**Comment by howardjohn** [comment]
If we did support this how would you want to configure this? Store the cert in 2 Secrets and reference both?

**Comment by rasvi24** [comment]
Yes. If tls field is converted from map to array to accept two certs it will be great. I prefer credential names so we don't have to manually mount them.

**Comment by howardjohn** [comment]
Ack.

http://gateway-api.org/ has support for this by using a list of secret names, for this exact case. So I think adding support is pretty reasonable. Unfortunately we cannot make it a list backwards compatibly, but we can add a new field to allow a list.
